command/server.go

```diff
@@ -446,9 +446,31 @@ func (c *ServerCommand) parseConfig() (*server.Config, []configutil.ConfigError,
 		c.UI.Warn("WARNING: Entropy Augmentation is not supported in FIPS 140-2 Inside mode; disabling from server configuration!\n")
 		config.Entropy = nil
 	}
-
 	entCheckRequestLimiter(c, config)
 
+	configsPresent := config != nil && config.SharedConfig != nil && config.Storage != nil
+
+	// ensure that the DisableMlock key is explicitly set if using integrated storage
+	if configsPresent &&
+		// using integrated storage
+		config.Storage.Type == storageTypeRaft &&
+		// DisableMlock key has been found and thus explicitly set
+		!(strutil.StrListContainsCaseInsensitive(config.SharedConfig.FoundKeys, "DisableMlock") ||
+			// mlock is disabled and hence has been explicitly set
+			config.SharedConfig.DisableMlock) {
+
+		c.UI.Error(wrapAtLength(
+			"ERROR: disable_mlock must be configured 'true' or 'false': Mlock " +
+				"prevents memory from being swapped to disk for security reasons, " +
+				"but can cause Vault on Integrated Storage to run out of memory " +
+				"if the machine was not provisioned with enough RAM. Disabling " +
+				"mlock can prevent this issue, but can result in the reveal of " +
+				"plaintext secrets when memory is swapped to disk, so it is only " +
+				"recommended on systems with encrypted swap.",
```

Comment:
Encrypted swap, or it's also fine if swap has been disabled, which is what our docs recommend.